core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/ngbd-modal-workflow-executions/ngbd-modal-workflow-executions.component.ts

```diff
@@ -146,4 +151,44 @@ export class NgbdModalWorkflowExecutionsComponent implements OnInit {
         );
       });
   }
+
+  /* Pagination handler */
+  onPageIndexChange(pageIndex: number): void {
+    this.currentPageIndex = pageIndex;
+    this.changePaginatedExecutions();
+  }
+
+  onPageSizeChange(pageSize: number): void {
+    this.pageSize = pageSize;
+    this.changePaginatedExecutions();
+  }
+
+  /**  Render executions when changing page
+   * To-do: create WorkflowExecutionEntry to handle pageIndex, or
+   * add page attributes to workflowExecutionService with function
+   * getPaginatedExecutions()
+   */
+  changePaginatedExecutions() {
+    if (this.workflow.wid === undefined) {
+      return;
+    }
+    this.workflowExecutionsService
+      .retrieveWorkflowExecutions(this.workflow.wid)
+      .pipe(untilDestroyed(this))
+      .subscribe(data => {
+        this.workflowExecutionsList = [];
```

Comment:
Why are you retrieving the list from the DB again? Why assign an empty array to the list?

Reply:
Sorry, I made a redundant code. displayWorkflowExecutions() will be removed to call api once by changePaginatedExecutions()